File: IntelliCare-Frontend/src/app/models/report-summary.dto.ts
export interface ReportSummaryDto {
  reportID: number;
  type: string;
  generatedDate: string;
  metrics: any[];
  detailedData?: any;
}
 
 

// export interface MetricDto {
//   label: string;
//   value: number;
//   unit: string;
// }

// export interface ReportSummaryDto {
//   reportID: number;
//   reportType: string;
//   generatedOn: string;
//   metrics: MetricDto[];
// }

File: IntelliCare.Sln/IntelliCare.Application/DTOs/ReportSummaryDto.cs
// IntelliCare.Application.DTOs/ReportSummaryDto.cs

using System;
using System.Collections.Generic;

namespace IntelliCare.Application.DTOs
{
    public class ReportSummaryDto
    {
        public int ReportID { get; set; }

        public string Type { get; set; } = string.Empty;
        public DateTime GeneratedDate { get; set; }

        public List<MetricDto> Metrics { get; set; } = new List<MetricDto>();
        public object? DetailedData { get; set; }
    }
}